src/status_im/multiaccounts/core_test.cljs

```diff
@@ -0,0 +1,85 @@
+(ns status-im.multiaccounts.core-test
+  (:require [cljs.test :refer-macros [deftest is testing]]
+            [status-im.multiaccounts.core :as ma]))
+
+(def pk "0x0461f576da67dc0bca9888cdb4cb28c80285b756b324109da94a081585ed6f007cf00afede6b3ee5638593674fee100b590318fc7bdb0054b8dd9445acea216ad2")
+(def short-pk "0x0461…ad2")
+(def random-name "Studious Gold Mustang")
+(def override-random-name (str "override" random-name))
+(def nickname "nickname")
+(def override-nickname (str nickname "override"))
+(def display-name "display-name")
+(def override-display-name (str display-name "override"))
+(def ens-name "jakubgs.eth")
+(def formatted-ens (str "@" ens-name))
+(def override-ens-name (str "override" ens-name))
+
+(def contact
+  {:nickname     nickname
+   :name         ens-name
+   :ens-verified true
+   :public-key   pk
+   :display-name display-name})
+
+(deftest contact-two-names
+  (testing "names is nil"
+    (testing "nickname has precedence"
+      (is
+       (= [nickname formatted-ens]
+          (ma/contact-two-names contact pk))))
+    (testing "ens name is second option"
+      (is
+       (= [formatted-ens display-name]
+          (ma/contact-two-names
+           (dissoc contact :nickname)
+           pk))))
+    (testing "ens name is second option but not verified"
+      (is
+       (= [display-name random-name]
+          (ma/contact-two-names
+           (dissoc contact :nickname :ens-verified)
+           pk))))
+    (testing "display name is third option"
+      (is
+       (= [display-name random-name]
+          (ma/contact-two-names
+           (dissoc contact :nickname :name)
+           pk))))
+    (testing "3 random words is fallback"
+      (is
+       (= [random-name short-pk]
+          (ma/contact-two-names
+           (dissoc contact
+                   :nickname
+                   :name
+                   :display-name)
+           pk)))))
+  (testing "names is provided"
+    (let [names              {:nickname         override-nickname
+                              :display-name     override-display-name
+                              :three-words-name override-random-name
+                              :ens-name         override-ens-name}
+          contact-with-names (assoc contact :names names)]
+
+      (testing "nickname has precedence"
```

Comment:
I am not sure if ens name should have precedence based on the designs. We can modify this later if needed btw, just raising my doubt.

Reply:
I'll double check with the design team, if that's the case I'll change it